components/automate-ui/src/app/entities/layout/layout-sidebar.service.ts

```diff
@@ -1,294 +1,321 @@
-                ],
-                visible: true
-            },
-            {
-                name: 'Identity',
-                items: [
-                    {
-                        name: 'Users',
-                        icon: 'person',
-                        route: '/settings/users',
-                        authorized: {
-                            name: 'users',
-                            allOf: ['/auth/users', 'get']
-                        },
-                        visible: true
-                    },
-                    {
-                        name: 'Teams',
-                        icon: 'people',
-                        route: '/settings/teams',
-                        authorized: {
-                            name: 'teams',
-                            allOf: ['/auth/teams', 'get']
-                        },
-                        visible: true
-                    },
-                    {
-                        name: 'API Tokens',
-                        icon: 'vpn_key',
-                        route: '/settings/tokens',
-                        authorized: {
-                            name: 'tokens',
-                            anyOf: [['/auth/tokens', 'get'],
-                            ['/iam/v2/tokens', 'get']]
-                        },
-                        visible: true
-                    }
-                ],
-                visible: true
-            },
-            {
-                name: 'Access Management',
-                items: [
-                    {
-                        name: 'Policies',
-                        icon: 'security',
-                        route: '/settings/policies',
-                        authorized: {
-                            name: 'policies',
-                            allOf: ['/iam/v2/policies', 'get']
-                        },
-                        visible: true
-                    },
-                    {
-                        name: 'Roles',
-                        icon: 'assignment_ind',
-                        route: '/settings/roles',
-                        authorized: {
-                            name: 'roles',
-                            allOf: ['/iam/v2/roles', 'get']
-                        },
-                        visible: true
-                    },
-                    {
-                        name: 'Projects',
-                        icon: 'work',
-                        route: '/settings/projects',
-                        authorized: {
-                            name: 'projects',
-                            allOf: ['/iam/v2/projects', 'get']
-                        },
-                        visible: true
-                    }
-                ],
-                visible: this.isIAMv2$
+    ngOnInit() {
+      this.updateSidebars();
+    }
+
+    ngOnDestroy() {
+      this.isDestroyed.next(true);
+      this.isDestroyed.complete();
+    }
+
+    private updateVisibleValues(sidebar): Sidebars {
```

Comment:
what type is `sidebar`?

Reply:
@bcmdarroch It's type is Sidebars but it's not the correct model when it gets passed into the function as the function is actually manipulating the object to match the Sidebars model and returning it back.